lib/logflare_web/controllers/plugs/ndjson_parser.ex

```diff
@@ -0,0 +1,71 @@
+defmodule Plug.Parsers.NDJSON do
+  @moduledoc """
+  Parse Cloudflare Log Push request bodies.
+  """
+  require Logger
+
+  @behaviour Plug.Parsers
+  import Plug.Conn
+  @gzip_header {"content-encoding", "gzip"}
+
+  def init(_params) do
+  end
+
+  def parse(conn, "application", "x-ndjson", _headers, _opts) do
+    conn
+    |> read_body()
+    |> decode()
+  end
+
+  @doc false
+  def parse(conn, _type, _subtype, _headers, _opts) do
+    {:next, conn}
+  end
+
+  def decode({:ok, <<>>, conn}) do
+    {:ok, %{}, conn}
+  end
+
+  def decode({:ok, body, conn}) do
+    body =
+      if @gzip_header in conn.req_headers do
+        body |> :zlib.gunzip() |> String.split("\n", trim: true)
+      else
+        body |> String.split("\n", trim: true)
+      end
+
+    batch =
+      for line <- body do
+        case Jason.decode(line) do
+          {:ok, log_event} ->
+            %{
+              "timestamp" => log_event["EdgeStartTimestamp"],
+              "message" => line,
+              "metadata" => log_event
+            }
+
+          {:error, error} ->
+            Logger.warn(inspect(error))
+
+            nil
```

Comment:
should we filter out nils? will likely lead to ingestion error for the entire batch

Reply:
I saw that in one of the other parsers but was trying not to have to Enum over the list if I didn't need to.

I will do that though just in case.